src/cosalib/aliyun.py

```diff
@@ -82,11 +89,49 @@ def aliyun_run_ore_replicate(build, args):
             } for region, val in ore_data.items()
         ])
 
+    # short circuit the rest of this: if we haven't uploaded to all the
+    # specified regions, we shouldn't be continuing to try to make them public
+    # or writing out the updated meta.json
+    if upload_failed_in_region is not None:
+        raise Exception(f"Upload failed in {upload_failed_in_region} region")
```

Comment:
So we exit here early and don't write to the meta.json if we failed to upload in any region. I assume the copy image is idempotent and if we run this again it will just do the right thing?

Reply:
I assumed that as well, but it looks like if you try to re-run `cosa aliyun-replicate` with this change, it'll throw an error.  Ultimately, `CopyImage()` complains the image name is duplicated.

I'll reorder this so that the `meta.json` will be written with the regions that were successful.  However, any failure to copy to a region will still short-circuit the attempt to make the images public,

Reply:
Yup!